addons/pos_self_order/static/src/ProductList/ProductList.js

```diff
@@ -0,0 +1,217 @@
+                    if (amountScrolled < threshold) {
+                        ticking = false;
+                        return;
+                    }
+                    scrollingDown = scrollY > lastScrollY;
+                    if (scrollingDown === this.privateState.navbarIsShown) {
+                        this.hideNavBar(this.privateState.navbarIsShown);
+                    }
+                    lastScrollY = scrollY > 0 ? scrollY : 0;
+                    ticking = false;
+                };
+
+                const onScroll = () => {
+                    if (!ticking) {
+                        window.requestAnimationFrame(updateScrollDir);
+                        ticking = true;
+                    }
+                };
+                this.productListDiv.el.addEventListener("scroll", onScroll);
+                return () => this.productListDiv.el.removeEventListener("scroll", onScroll);
+            },
+            () => [this.privateState.scrollingDown, this.privateState.searchIsFocused]
+        );
+        // this IntersectionObserver is used to highlight the tag (in the header)
+        // of the category that is currently visible in the viewport
+        useEffect(
+            (searchIsFocused) => {
+                if (searchIsFocused) {
+                    return;
+                }
+                const observer = new IntersectionObserver(
+                    (entries) => {
+                        const entry = entries.filter((entry) => entry.isIntersecting)?.[0];
+                        if (entry) {
+                            this.privateState.selectedTag =
+                                entry.target.querySelector("span").innerHTML;
+                            this.tagList?.el?.scroll({
+                                top: 0,
+                                left:
+                                    this.tagButtons[this.privateState.selectedTag]?.el.offsetLeft -
+                                    window.innerWidth / 2,
+                                behavior: "smooth",
+                            });
+                        }
+                    },
+                    {
+                        root: this.productListDiv?.el,
+                        rootMargin: `0px 0px -${this.productListDiv.el.offsetHeight - 5}px 0px`,
+                    }
+                );
+                Object.keys(this.productGroupDiv).forEach((tag) => {
+                    observer.observe(this.productGroupDiv[tag]?.el);
+                });
+                return () => {
+                    observer.disconnect();
+                };
+            },
+            () => [this.privateState.searchIsFocused]
+        );
+    }
+    /**
+     * @param {boolean} hide - true if the navbar should be hidden, false otherwise
+     * @description this function hides or shows the navbar by sliding the whole productPage up or down
+     */
+    hideNavBar = (hide) => {
```

Comment:
```suggestion
    hideNavBar(hide) {
```
Don't declare the methods as arrow functions.